test/jdk/java/net/httpclient/MappingResponseSubscriber.java

```diff
@@ -126,15 +130,31 @@ HttpClient newHttpClient() {
     public void testAsBytes(String uri, boolean sameClient) throws Exception {
         HttpClient client = null;
         for (int i = 0; i < ITERATION_COUNT; i++) {
-            if (!sameClient || client == null)
-                client = newHttpClient();
-
-            HttpRequest req = HttpRequest.newBuilder(URI.create(uri))
-                                         .build();
-            BodyHandler<byte[]> handler = new CRSBodyHandler();
-            HttpResponse<byte[]> response = client.send(req, handler);
-            byte[] body = response.body();
-            assertEquals(body, bytes);
+            try {
+                if (!sameClient || client == null)
+                    client = newHttpClient();
+
+                HttpRequest req = HttpRequest.newBuilder(URI.create(uri))
+                        .build();
+                BodyHandler<byte[]> handler = new CRSBodyHandler();
+                HttpResponse<byte[]> response = client.send(req, handler);
+                byte[] body = response.body();
+                assertEquals(body, bytes);
+            } finally {
+                if (sameClient) continue;
+                Tracker tracker = TRACKER.getTracker(client);
+                client = null;
+                System.gc();
+                AssertionError error = TRACKER.check(tracker, 1500);
+                if (error != null) throw error;
```

Comment:
Hello Daniel, in theory, this could potentially lose the any original failure/exception that the test in the try block might have raised. Perhaps we could catch and throw the original exception and add this additional failure as a suppressed exception to it?

Reply:
Good point! Let me fix that.